### PR TITLE
Add Search Sort Ordering

### PR DESCRIPTION
**Pull request by joel-wright**
Sorting now supported ascending/descending according to:
- score
- date
- title

### CODE REVIEW

File: backend/app/api/api_v1/routers/tests/test_search.py [hunk 1]
```diff
@@ -140,11 +137,126 @@ def test_multiple_filters(
     } in query_body["query"]["bool"]["filter"]
 
 
-def test_result_order(test_opensearch, monkeypatch, client, user_token_headers):
+@pytest.mark.parametrize("order", [SortOrder.DESCENDING, SortOrder.DESCENDING])
```
Comment (kdutia) [inline backend/app/api/api_v1/routers/tests/test_search.py:140]: I think this should be `[SortOrder.DESCENDING, SortOrder.ASCENDING]`

Comment (kdutia) [inline backend/app/api/api_v1/routers/tests/test_search.py:140]: Do we need `DESCENDING` twice here?

File: backend/app/api/api_v1/routers/tests/test_search.py [hunk 2]
```diff
@@ -140,11 +137,126 @@ def test_multiple_filters(
...
+        new_s = d["top_hit"]["value"]
+        if s is not None:
+            if order == SortOrder.DESCENDING:
+                assert new_s <= s
+            if order == SortOrder.ASCENDING:
+                assert new_s >= s
+        s = new_s
+
+
+@pytest.mark.parametrize("order", [SortOrder.DESCENDING, SortOrder.DESCENDING])
```
Comment (kdutia) [inline backend/app/api/api_v1/routers/tests/test_search.py:171]: Think this should be ASCENDING, DESCENDING. Same with `test_result_order_title` below (but not the score one).

File: backend/app/db/schemas/search.py [hunk 3]
```diff
@@ -35,7 +34,7 @@ class SearchRequestBody(BaseModel):
     year_range: Optional[Tuple[Optional[int], Optional[int]]] = None
 
     sort_field: SortField = SortField.DATE
```
Comment (kdutia) [inline backend/app/db/schemas/search.py:36]: Does this mean that the default sort order of search results is date descending? It should be score descending, right?

**Comment by kdutia** [comment]
Looks good! I couldn't leave a comment inline as the line wasn't changed in this PR, but I think the `stats` clause in the Opensearch query could be amended as follows, as that clause's only purpose is to enable sorting by date.

``` diff
"action_date": {
    "stats": {
-        "field": "action_date",
+        "field": _SORT_FIELD_MAP[SortField.DATE],
    },
},
```

File: backend/app/api/api_v1/routers/tests/test_search.py (hunk 1 above)
Comment (joel-wright) [reply backend/app/api/api_v1/routers/tests/test_search.py:140]: :man_facepalming:

File: backend/app/db/schemas/search.py (hunk 3 above)
Comment (joel-wright) [reply backend/app/db/schemas/search.py:36]: I really should keep a more careful eye on autocomplete

**Pull request closed without merging**